nupic/research/spatial_pooler.py

```diff
@@ -717,25 +710,26 @@ def compute(self, inputVector, learn, activeArray, stripNeverLearned=True):
       if self._isUpdateRound():
         self._updateInhibitionRadius()
         self._updateMinDutyCycles()
-    elif stripNeverLearned:
-      activeColumns = self.stripUnlearnedColumns(activeColumns)
 
     activeArray.fill(0)
     if activeColumns.size > 0:
       activeArray[activeColumns] = 1
 
 
-  def stripUnlearnedColumns(self, activeColumns):
+  def stripUnlearnedColumns(self, activeArray):
     """Removes the set of columns who have never been active from the set of
     active columns selected in the inhibition round. Such columns cannot
     represent learned pattern and are therefore meaningless if only inference
     is required. This should not be done when using a random, unlearned SP
     since you would end up with no active columns.
 
-    @param activeColumns: An array containing the indices of the active columns
+    @param activeArray: An array whose size is equal to the number of columns.
+        Each column in this array will be checked to see if it has been active
+        before. If a column is active but it hasn't been learned, it will be
+        disabled.
     """
     neverLearned = numpy.where(self._activeDutyCycles == 0)[0]
```

Comment:
I see answer to my Q above. Maybe better mention this "learned <=> activeDutyCycles > 0"